libc/stdio_scanf.c

```diff
@@ -0,0 +1,3506 @@
+
+	TEST_ASSERT_EQUAL_INT64(LLONG_MIN, int_min);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MAX, int_max);
+	TEST_ASSERT_EQUAL_INT64(0, int_zero);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MIN / 2, int_half_min);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MAX / 2, int_half_max);
+
+	TEST_ASSERT_EQUAL_INT(5, vread_array(FreeSpace, "%lld %lld %lld %lld %lld", &int_max, &int_half_max, &int_zero, &int_half_min, &int_min));
+
+	TEST_ASSERT_EQUAL_INT64(LLONG_MIN, int_min);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MAX, int_max);
+	TEST_ASSERT_EQUAL_INT64(0, int_zero);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MIN / 2, int_half_min);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MAX / 2, int_half_max);
+
+	TEST_ASSERT_EQUAL_INT(5, sscanf(FreeSpace, "%lld %lld %lld %lld %lld", &int_max, &int_half_max, &int_zero, &int_half_min, &int_min));
+
+	TEST_ASSERT_EQUAL_INT64(LLONG_MIN, int_min);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MAX, int_max);
+	TEST_ASSERT_EQUAL_INT64(0, int_zero);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MIN / 2, int_half_min);
+	TEST_ASSERT_EQUAL_INT64(LLONG_MAX / 2, int_half_max);
+}
+
+
+TEST(SCANF_EXT_D, JD)
+{
+	char FreeSpace[250];
+	intmax_t int_max, int_min, int_zero, int_half_min, int_half_max;
+
+	sprintf(FreeSpace, "%ld %ld %d %ld %ld", LONG_MAX, LONG_MAX / 2, 0, LONG_MIN / 2, LONG_MIN);
+	create_file(std_path, FreeSpace);
+	rd = fopen(std_path, "r+");
+
+	TEST_ASSERT_EQUAL_INT(5, vread_file(rd, "%jd %jd %jd %jd %jd", &int_max, &int_half_max, &int_zero, &int_half_min, &int_min));
+
+	TEST_ASSERT_EQUAL_INT64(LONG_MIN, int_min);
+	TEST_ASSERT_EQUAL_INT64(LONG_MAX, int_max);
+	TEST_ASSERT_EQUAL_INT64(0, int_zero);
+	TEST_ASSERT_EQUAL_INT64(LONG_MIN / 2, int_half_min);
+	TEST_ASSERT_EQUAL_INT64(LONG_MAX / 2, int_half_max);
+
+	rewind(rd);
+
+	TEST_ASSERT_EQUAL_INT(5, fscanf(rd, "%jd %jd %jd %jd %jd", &int_max, &int_half_max, &int_zero, &int_half_min, &int_min));
+
+	TEST_ASSERT_EQUAL_INT64(LONG_MIN, int_min);
+	TEST_ASSERT_EQUAL_INT64(LONG_MAX, int_max);
+	TEST_ASSERT_EQUAL_INT64(0, int_zero);
+	TEST_ASSERT_EQUAL_INT64(LONG_MIN / 2, int_half_min);
+	TEST_ASSERT_EQUAL_INT64(LONG_MAX / 2, int_half_max);
+
+	TEST_ASSERT_EQUAL_INT(5, vread_array(FreeSpace, "%jd %jd %jd %jd %jd", &int_max, &int_half_max, &int_zero, &int_half_min, &int_min));
+
+	TEST_ASSERT_EQUAL_INT64(LONG_MIN, int_min);
+	TEST_ASSERT_EQUAL_INT64(LONG_MAX, int_max);
+	TEST_ASSERT_EQUAL_INT64(0, int_zero);
+	TEST_ASSERT_EQUAL_INT64(LONG_MIN / 2, int_half_min);
+	TEST_ASSERT_EQUAL_INT64(LONG_MAX / 2, int_half_max);
+
+	TEST_ASSERT_EQUAL_INT(5, sscanf(FreeSpace, "%jd %jd %jd %jd %jd", &int_max, &int_half_max, &int_zero, &int_half_min, &int_min));
+
+	TEST_ASSERT_EQUAL_INT64(LONG_MIN, int_min);
+	TEST_ASSERT_EQUAL_INT64(LONG_MAX, int_max);
+	TEST_ASSERT_EQUAL_INT64(0, int_zero);
+	TEST_ASSERT_EQUAL_INT64(LONG_MIN / 2, int_half_min);
+	TEST_ASSERT_EQUAL_INT64(LONG_MAX / 2, int_half_max);
+}
+
+
+TEST(SCANF_EXT_D, ZD)
+{
+	char FreeSpace[400];
+	size_t int_max, int_zero, int_half_max;
+
+	sprintf(FreeSpace, "%zd %zd %d", SSIZE_MAX, SSIZE_MAX / 2, 0);
```

Comment:
Please resolve the following warnings: I think the best option there is to cast. As it's described in PR's checklist we don't want to increase the number of warnings. Now you can check all warnings related to your code in PR->Files changed 

![Screenshot from 2023-03-03 13-04-33](https://user-images.githubusercontent.com/77304431/222716390-f4b184a9-07bc-4638-8241-03982c9069be.png)